runagent/utils/enums/framework.py

```
@@ -0,0 +1,157 @@
+    def get_selectable_frameworks(cls) -> t.List['Framework']:
+        """Get frameworks that can be selected (excluding DEFAULT)"""
+        return [f for f in cls if f != cls.DEFAULT]
+
+    @classmethod
+    def from_string(cls, value: str) -> 'Framework':
+        """Convert string to Framework with validation"""
+        try:
+            return cls(value)
+        except ValueError:
+            valid_values = [f.value for f in cls]
+            raise ValueError(f"Invalid framework: '{value}'. Valid options: {valid_values}")
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)